src/controllers/reviewController.js

```diff
@@ -0,0 +1,112 @@
+        if (existingReview) {
+          return res.status(409).json({ status: "fail", message: "You have already reviewed this product." });
+        }
+        const review = await Review.create({
+          productId,
+          userId,
+          title,
+          content,
+          rating,
+          
+        });
+        return res.status(201).json({ status: "success", review });
+
+      } catch (error) {
+        return res.status(500).json({ status: "fail", error: error.message });
+      }
+    }
+  
+    static async updateReview(req, res) {
+      try {
+        const { rating, content } = req.body;
+        const reviewId = req.params.id;
+        const userId = req.user.id;
+        const existingReview = await Review.findOne({ where: { id: reviewId, userId } });
+        if (!existingReview) {
+          return res.status(404).json({ status: "fail", message: "Review not found." });
+        }
+        const review = await existingReview.update({ rating, content });
+        return res.status(200).json({ status: "success", review });
+      } catch (error) {
+        return res.status(500).json({ status: "fail", error: error.message });
+      }
+    }
+  
+    static async deleteReview(req, res) {
+      try {
+        const reviewId = req.params.id;
+        const userId = req.user.id;
+        const existingReview = await Review.findOne({ where: { id: reviewId, userId } });
+        if (!existingReview) {
+          return res.status(404).json({ status: "fail", message: "Review not found." });
+        }
+        await existingReview.destroy();
+        return res.status(204).json({ status: "success", message: "Review deleted." });
+      } catch (error) {
+        return res.status(500).json({ status: "fail", error: error.message });
+      }
+    }
+  
+    static async getProductReviews(req, res) {
+      try {
+        const productId = req.params.id;
+        const existingProduct = await Product.findByPk(productId);
+        if (!existingProduct) {
+          return res.status(404).json({ status: "fail", message: "Product not found." });
+        }
+        const reviews = await Review.findAll({
+          where: { productId },
+          include: [{ model: User, attributes: ["firstName", "lastName"] }]
+        });
+        return res.status(200).json({ status: "success", data: reviews });
+      } catch (error) {
+        return res.status(500).json({ status: "fail", error: error.message });
+      }
+    }
+  
+
+
+  static async getProductRate(req, res) {
+    const productId = req.params.id;
+    const product = await Product.findByPk(productId);
+  
+    if (!product) {
+      return res.status(404).json({ error: 'Product not found' });
+    }
+  
+    // get product reviews and calculate average rating
+    const reviews = await product.getReviews();
+    const totalRating = reviews.reduce((sum, review) => sum + review.rating, 0);
+    const averageRating = totalRating / reviews.length;
+  
+    res.json({
```

Comment:
- good job on reviews and rating overall but on this response try to return meaningful and correct response as you did in above lines.